### PR TITLE
10074 bug: Auto Scroll Issues on Petitions Clerk Create Case Form (and more)

### PR DESCRIPTION
**Pull request by Mwindo**
TL;DR: Fixing ~~some~~ quite a few form validation bugs/quirks.

---

The goals of this PR:
- Fix a couple of bugs
- When applicable, update `ErrorNotification` live to keep it in sync with in-line validations. (The previous behavior had `ErrorNotification` only updating on submissions, not validations.)
- Refactor to make validation slightly more consistent across the app and to eliminate duplicative actions (namely, `setAlertErrorAction` followed by `setValidationAlertErrorsAction`).

In the future, we will probably move away from the `ErrorNotification` altogether--similar to how things work with the new petitioner flow--but this at least keeps behavior consistent until then and hopefully will make a refactor easier. 

---

An outline of the issues addressed:

1. After (almost) every field edit on the petitions clerk create case form, we were scrolling back to the `ErrorNotification` banner. (This was the original error of [10074](https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/gh/flexion/ef-cms/10074)).
~~2. In certain cases on the petitions clerk create case form, the form scrolled weirdly. (This was design debt [10357](https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/gh/flexion/ef-cms/10357).)~~ Not fixed!
3. In the `PetitionsPayment` forms, blank Payment Dates and Payment Methods led to 400 errors rather than validation errors. (See video below.)

https://github.com/user-attachments/assets/f6888b74-2aa3-4bea-b6ae-90ed7cdf33e7

4. Certain onBlurs/onChanges weren't providing live validation. (E.g., Status Report Order validation was not providing live validation updates.) There are probably still some that I have missed.

### CODE REVIEW

File: web-client/src/presenter/actions/validateCaseDetailsAction.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ export const validateCaseDetailsAction = async ({
         caseDetail: {
           ...caseDetail,
           ...form,
+          petitionPaymentDate: form.petitionPaymentDate || null,
+          petitionPaymentMethod: form.petitionPaymentMethod || null,
```
Comment (Mwindo) [inline web-client/src/presenter/actions/validateCaseDetailsAction.ts:23]: This fixes issue 2 as listed in the PR description by triggering the validation rather than allowing the form to pass and then cause a 400.

File: web-client/src/presenter/actions/validatePetitionerAction.ts [hunk 2]
```diff
@@ -26,11 +25,9 @@ export const validatePetitionerAction = ({
       existingPetitioners: caseDetail.petitioners,
     });
 
-  store.set(state.validationErrors.contact, errors);
-
   if (isEmpty(errors)) {
     return path.success();
   } else {
-    return path.error({ errors });
+    return path.error({ errors: { contact: errors } });
```
Comment (Mwindo) [inline web-client/src/presenter/actions/validatePetitionerAction.ts:31]: The way this action was functioning was failing to update the corresponding form (`EditPetitionerInformationInternal.tsx`) appropriately.

File: web-client/src/presenter/computeds/alertHelper.ts [hunk 3]
```diff
@@ -15,7 +15,6 @@ export const alertHelper = (get: Get): any => {
 
   return {
     messagesDeduped: uniq(alertError.messages).filter(Boolean),
-    preventAutoScroll: false,
```
Comment (Mwindo) [inline web-client/src/presenter/computeds/alertHelper.ts:18]: This is useless since it is always false.

File: web-client/src/presenter/actions/setValidationAlertErrorsAction.ts [hunk 4]
```diff
@@ -86,6 +86,7 @@ export const setValidationAlertErrorsAction = ({
           }
         }),
     ),
+    scrollToErrorNotification: props.scrollToErrorNotification || false,
```
Comment (Mwindo) [inline web-client/src/presenter/actions/setValidationAlertErrorsAction.ts:89]: This gives us a way explicitly telling the app when to scroll to the `ErrorNotification` banner. By default, we don't scroll. To change the default, we can call the above `setScrollToErrorNotificationAction`.

File: web-client/src/presenter/sequences/JudgeActivityReport/submitJudgeActivityReportSequence.ts [hunk 5]
```diff
@@ -27,8 +27,8 @@ export const submitJudgeActivityReportSequence = showProgressSequenceDecorator([
   validateJudgeActivityReportSearchAction,
   {
     error: [
-      setAlertErrorAction,
```
Comment (Mwindo) [inline web-client/src/presenter/sequences/JudgeActivityReport/submitJudgeActivityReportSequence.ts:30]: Throughout, I have removed `setAlertErrorAction` when it precedes `setValidationAlertErrorsAction` since the latter overwrites the former. On submissions, I have called `setScrollToErrorNotificationsAction`. A good portion of this PR is refactoring sequences to follow this pattern.

File: shared/src/business/entities/cases/Case.ts [hunk 6]
```diff
@@ -534,7 +534,7 @@ export class Case extends JoiValidationEntity {
     mailingDate: JoiValidationConstants.STRING.max(25)
       .when('isPaper', {
         is: true,
-        otherwise: joi.allow(null).optional(),
+        otherwise: joi.optional().allow(null),
```
Comment (Mwindo) [inline shared/src/business/entities/cases/Case.ts:537]: Just making the order of these consistent in the file.

File: web-client/src/presenter/actions/setValidationAlertErrorsAction.test.ts [hunk 7]
```diff
@@ -140,4 +141,23 @@ describe('setValidationAlertErrors', () => {
       messages: ['second nested', 'first nested'],
     });
   });
+
+  it('state.alertError sets scrollToErrorNotification appropriately', async () => {
```
Comment (En-8) [inline web-client/src/presenter/actions/setValidationAlertErrorsAction.test.ts]: nit: this doesn't really read like a sentence ("it does x [when y]") the same way our other test cases do.

**Review by En-8 — APPROVED** [review]
LGTM :)

File: web-client/src/presenter/actions/setValidationAlertErrorsAction.test.ts (hunk 7 above)
Comment (Mwindo) [reply web-client/src/presenter/actions/setValidationAlertErrorsAction.test.ts]: You're right--I will update! Thanks :)
